deeplearning4j/deeplearning4j-ui-parent/deeplearning4j-vertx/src/main/java/org/deeplearning4j/ui/VertxUIServer.java

```diff
@@ -188,19 +189,17 @@ private static void deploy(Promise<String> startCallback) {
         Vertx vertx = Vertx.vertx();
         vertx.deployVerticle(VertxUIServer.class.getName(), promise);
 
-        Thread currentThread = Thread.currentThread();
-        VertxUIServer.autoStopThread = new Thread(() -> {
-            try {
-                currentThread.join();
-                if (VertxUIServer.instance != null && !VertxUIServer.instance.isStopped()) {
-                    log.info("Deeplearning4j UI server is auto-stopping after thread (name: {}) died.",
-                            currentThread.getName());
+        VertxUIServer.shutdownHook = new Thread(() -> {
+            if (VertxUIServer.instance != null && !VertxUIServer.instance.isStopped()) {
+                log.info("Deeplearning4j UI server is auto-stopping in shutdown hook.");
+                try {
                     instance.stop();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```

Comment:
You may want to log this.

Reply:
Yes, this should be logged. I will change this.